src/client/gameannouncer.py

```diff
@@ -1,61 +1,86 @@
             return
-        self._announce(game, "hosting")
+        i = 0
+        for event in self._delayed_event_list:
+            if player in event:
+                player, event = self._delayed_event_list.pop(i)
+                self._friend_announce(player, event)
+            i += 1
 
-    def _announce_replay(self, game):
-        if not self._is_friend_host(game) or not self.announce_replays:
+    def _friend_announce(self, player, event):
+        if player.currentGame is None:
+            return
+        game = player.currentGame
+        if event == FriendEvents.HOSTING_GAME:
+            if not self.announce_games:  # Menu Option Chat
+                return
+            if game.featured_mod == "ladder1v1":
+                activity = "started"
+            else:
+                activity = "is <font color='GoldenRod'>hosting</font>"
+        elif event == FriendEvents.JOINED_GAME:
+            if not self.announce_games:  # Menu Option Chat
+                return
+            if game.featured_mod == "ladder1v1":
+                activity = "started"
+            else:
+                activity = "joined"
+        elif event == FriendEvents.REPLAY_AVAILABLE:
+            if not self.announce_replays:  # Menu Option Chat
+                return
+            activity = "is playing live"
+        else:  # that shouldn't happen
             return
-        self._announce(game, "playing live")
 
-    def _announce(self, game, activity):
-        url = game.url(game.host_player.id).toString()
-        url_color = self._colors.getColor("url")
-        mapname = maps.getDisplayName(game.mapname)
-        fmt = 'is {} {}<a style="color:{}" href="{}">{}</a> (on {})'
         if game.featured_mod == "faf":
             modname = ""
         else:
             modname = game.featured_mod + " "
-        msg = fmt.format(activity, modname, url_color, url, game.title, mapname)
-        self._client.forwardLocalBroadcast(game.host, msg)
+        if game.featured_mod != "ladder1v1":
+            player_info = " [{}/{}]".format(game.num_players, game.max_players)
+        else:
+            player_info = ""
+        time_info = ""
+        if game.has_live_replay:
+            time_running = time.time() - game.launched_at
+            if time_running > 6 * 60:  # already running games on client start
+                time_format = '%M:%S' if time_running < 60 * 60 else '%H:%M:%S'
+                time_info = " runs {}"\
+                    .format(time.strftime(time_format, time.gmtime(time_running)))
+        url_color = self._colors.getColor("url")
+        url = game.url(player.id).toString()
+
+        fmt = '{} {}<a style="color:{}" href="{}">{}</a> ' \
+              '(on <font color="GoldenRod">{}</font> {}{})'
+        msg = fmt.format(activity, modname, url_color, url, game.title,
+                         game.mapdisplayname, player_info, time_info)
+        self._client.forwardLocalBroadcast(player.login, msg)
```

Comment:
For all I can tell, client's localBroadcast signal is only used by the chat widget. If we give this class its own signal that it emits here and pass the class to the chat widget in the constructor, we can get rid of an unnecessary indirection layer.

Reply:
There is an additional use of localBroadcast emit("Scores", message["text"]), in handle_notice in _clientwindow. (but I can't remember ever seen that message, is it in use, does it exist?)
That the GameAnnouncer can signal itself is a nice idea, but it is not a Qt object, so how signaling?

Reply:
We can add signals to GameAnnouncer exactly like to other objects, we just need to subclass it from QObject.